state/machine_linklayerdevices.go

```diff
@@ -206,18 +206,27 @@ func (m *Machine) SetLinkLayerDevices(devicesArgs ...LinkLayerDeviceArgs) (err e
 			}
 		}
 
+		// We've checked the model is alive directly, and we assert the machine is alive, we don't need to also
+		// assert the model is alive, because then the machine would be dying as well.
```

Comment:
I think you're right, and we should only care about checking model life when creating new resources.

IMO, the in-memory life check should be dropped too. It's of dubious value when there's no accompanying txn op, since it's no longer ensuring integrity - so it's really just an optimisation. This makes things more difficult to understand, because it's not clear why we would do it in only memory from the context (hence this question).

FWIW, I've taken to the pattern of:

```go
// checkSomething checks something in memory, and then return txn.Ops that assert the same
func checkSomething() ([]txn.Op, error)
```

That way it's clear that the ops will be ignored or not, more likely with comments to justify.